sam.c

```diff
@@ -1162,7 +1164,8 @@ enum SamError sam_cmd(Vis *vis, const char *s) {
 				break;
 			}
 		}
-		vis_mode_switch(vis, completed ? VIS_MODE_NORMAL : VIS_MODE_VISUAL);
+		if (!completed)
+			vis_mode_switch(vis, VIS_MODE_VISUAL);
```

Comment:
Still not to sure about this. It passes the tests, but I did not take time to investigate fully.
There is a potential issue here with recursive events being fired on mode switch